src/XrdHttp/XrdHttpReq.cc

```diff
@@ -585,14 +585,12 @@ bool XrdHttpReq::Redir(XrdXrootd::Bridge::Context &info, //!< the result context
     redirdest += buf;
   }
 
-  redirdest += quote(resource.c_str());
+  redirdest += encode_str<std::string>(resource.c_str()).c_str();
```

Comment:
This looks fishy, `encode_str<std::string>`, but with `resource.c_str()`, which is a `char*`, then `.c_str()` at the end again to get from `std::string` to `char*`. Maybe some optimization is possible here?

Reply:
It's because redirdest is a XrdOucString, it allows me not to have to free the raw pointer afterwards...